### DISCUSSION TITLE
Migrating the Existing app to Remix

### DISCUSSION THREAD

**Discussion by ram2104** [remix-run/remix#3435] [Show and tell / tips]
We've thinking to migrate one of (React, Redux, and React-Router) based apps to move into Remix. Because remix does support the incremental migration for any react-router-based application.

For the migration, I was following this documentation: https://remix.run/docs/en/v1/guides/migrating-react-router-app 

With the help of the documentation, I was able to bring the UI of the app on screen, but in general, all the logic related to the redux store is failing. When I look for the redux store it doesn't have anything. 

How can we migrate the app gracefully with redux? We can't drastically switch to loaders & actions because our codebase is huge.

**Comment by sergiodxa** [comment]
I would recommend you to first upgrade to React Router v6.4 (currently in [beta](https://beta.reactrouter.com/en/dev)) and gradually move your app to loader and actions there (that version of RR supports loaders and actions defined in the Route component.

Once you have done that, switching to Remix should be way simpler, and if in the upgrade to v6.4 you use the same file organization for your routes as Remix uses you can probably move the file directly to a Remix app and do small changes to adapt to SSR.

**Reply by ram2104** [reply]
Thanks, @sergiodxa  for the suggestion.
 I was able to resolve the issue by upgrading the modules.
